### DISCUSSION TITLE
Gatsby's redux store is required in the "addRemoteFilePolyfillInterface" and "polyfillImageServiceDevRoutes" API's for image CDN

### DISCUSSION THREAD

**Discussion by byjacobbyers** [gatsbyjs/gatsby#36080] [Help] [answered]
I'm facing a new warning in regards to Headless WordPress & Gatsby. 

```
Gatsby's redux store is required in the "addRemoteFilePolyfillInterface"
and "polyfillImageServiceDevRoutes" API's for image CDN. No store was found when
 requesting url "https://www.website.com/wp-content/uploads/2021/04/
slack_integration.png". This will fail your build in Gatsby V5. Upgrade your
source plugins and Gatsby packages to the latest versions to resolve this. If
you are a source plugin author visit https://gatsby.dev/source-plugins-image-cdn
 to learn how to make this warning go away.
```
This image is served from WordPress in a Gutenberg block to Gatsby via [`InnerHTML` ](https://www.npmjs.com/package/dangerously-set-html-content )so I am not really sure why Redux is needed for this. Is there a different way I need to be serving images from WordPress Gutenberg Blocks?

**Comment by bonmaklad** [comment]
same issue. Looks like we need to update react and react-dom but I will try tomorrow.

**Comment by oleg-boreiko-idf** [comment, marked as answer]
Fix this in version Gatsby 4.18.1 and gatsby-source-wordpress 6.18.1. Use `npm i gatsby@latest gatsby-source-wordpress@lateset`

**Reply by bonmaklad** [reply]
Weirdly enough I tried updating everything and now its doing it again a week later. Trying again now

**Reply by bonmaklad** [reply]
Nope. I went and deleted the picture it hated so much, which gets rid of the warning but still leaves me with:

ERROR 

UNHANDLED REJECTION pngload_buffer: libspng read error
vips2png: unable to write to target target

 WorkerError: pngload_buffer: libspng read error
  vips2png: unable to write to target target
  
  
  gatsby thought it might be react or react-dom updates, so I did those as well. I'll keep trying

**Comment by bonmaklad** [comment]
Ok I got a successful build locally and through Gatsby pipeline by deleting every image above 1meg and putting new images in its place. 

I did npm outdated and then npm i AddModule@latest on each one 

Then I went to wordpress, deleted all images from posts and pages

Then I deleted all images above 1meg on media 

then I went back and updated image with lower resolution images 

Worked first time, second time and now third time. 

no idea why... weird. Maybe a buffering issue on high res images?

**Reply by bonmaklad** [reply]
I tried adding back another high quality png and it failed. So yes, anything over 1meg becomes a hit and miss.

**Reply by bonmaklad** [reply]
Also I now notice one of my other bugs is gone. Whenever I would update any other page, it would cause the homepage to lose all its advanced fields. Forcing me to go back to the homepage and re-save. Thats now fixed aswell.

**Reply by bonmaklad** [reply]
Nope I lie :P if I update from github, its ok, if I update a post, thats ok, but if i update another page, it needs me to go back and save home. False alarm :P but issue one fixed :)

**Comment by byjacobbyers** [comment]
@bonmaklad I can try reducing image sizes over time to see if that fixes it for me too. I manage a site with a lot of images so it may take some time.

**Reply by bonmaklad** [reply]
I was lucky that I only had 5 images above 1meg. Very strange though, I did put a help desk request into Gatsby. I feel like this is something new

**Comment by byjacobbyers** [comment]
I was able to get our images down below 1MB and so far I am not seeing the warning. Granted there are several other tests I would need to make to confirm. I'll update here when I get more confirmation.

**Comment by byjacobbyers** [comment]
So after updating my source plugins the warning came back. And weirdly it's pinging me for a png file that's 300kb. I'll take a look into the template that the image is on next when I have a moment.
